libs/wingsdk/src/shared-tf/app.ts

```diff
@@ -40,7 +41,10 @@ export abstract class CdktfApp extends App {
     const cdktfApp = new cdktf.App({ outdir: cdktfOutdir });
     const cdktfStack = new cdktf.TerraformStack(cdktfApp, TERRAFORM_STACK_NAME);
 
-    super(cdktfStack, "Default");
+    const rootId = props.isTestEnvironment
+      ? "Default_" + nanoid(10)
```

Comment:
Random IDs for constructs should be strongly avoided. We want compilations to be deterministic and reproducible, so when you instantiate a class or call a method during preflight, the behavior should be the same each time as long as your environment is the same (e.g. same environment variables are set and so on).

How about we instead allow the root ID to be customized through AppProps or an environment variable? This way choosing the ID will be the responsibility of the CLI or user instead.

Reply:
If I understand correctly, this is only for test environments which are expected to be teared down anyway and be isolated from the environment. Kind of feels like a reasonable case for a random id, no?

Reply:
True @eladb :) 
I can also pass it from the user, the default will be just "Default", and in testing it can be automated to
 "<test folder>_<test file>". it may be better but still not collision-proof.... what do you think? @eladb @Chriscbr

Reply:
@eladb What about snapshot tests?

Reply:
> @eladb What about snapshot tests?

Good point. Here's a suggestion: let's add `--rootid` switch which will *default to nanoid* but snapshot tests can pass some fixed value there. WDYT?

Eventually we need snapshot testing to be built into `wing test`, so this should be handled internally.

Reply:
Added the switch, the snapshot tests are currently on the sim only, there I haven't changed the root name, since it's hardcoded in a few places in our code... (and we're not worried about tf not working well, the main reason to randomize the names- since we are not using tf for the sim)

Reply:
> since it's hardcoded in a few places in our code

Maybe open an issue to address it later?

Reply:
opened: https://github.com/winglang/wing/issues/3749 :)